src/common/io.ts

```diff
@@ -829,7 +905,7 @@ export class ARIOReadable implements AoARIORead {
 }
 
 export class ARIOWriteable extends ARIOReadable implements AoARIOWrite {
-  protected declare process: AOProcess;
+  public declare readonly process: AOProcess;
```

Comment:
nit: not actually sure why this needs to be defined. TypeScript seems to compile fine when removing
```suggestion
```

Reply:
older version of typescript complained, will review in a separate PR.